app/frontend/src/features/communities/events/hooks.ts

```diff
@@ -0,0 +1,58 @@
+import useUsers from "features/userQueries/useUsers";
+import { Error as GrpcError } from "grpc-web";
+import { ListEventAttendeesRes, ListEventOrganizersRes } from "proto/events_pb";
+import { eventAttendeesKey, eventOrganisersKey, QueryType } from "queryKeys";
+import { useInfiniteQuery } from "react-query";
+import { service } from "service";
+
+export interface UseEventUsersInput {
+  eventId: number;
+  type: QueryType;
+  pageSize?: number;
+}
+
+export function useEventOrganisers({
+  eventId,
+  pageSize,
+  type,
+}: UseEventUsersInput) {
+  const query = useInfiniteQuery<ListEventOrganizersRes.AsObject, GrpcError>({
+    queryKey: eventOrganisersKey({ eventId, type }),
+    queryFn: () => service.events.listEventOrganisers({ eventId, pageSize }),
+    getNextPageParam: (lastPage) => lastPage.nextPageToken || undefined,
+  });
+  const organiserIds =
+    query.data?.pages.flatMap((res) => res.organizerUserIdsList) ?? [];
+  const { data: organisers, isLoading: isOrganisersLoading } =
+    useUsers(organiserIds);
+
+  return {
+    ...query,
+    organiserIds,
+    organisers,
+    isLoading: query.isLoading || isOrganisersLoading,
+  };
+}
+
+export function useEventAttendees({
+  eventId,
+  pageSize,
+  type,
+}: UseEventUsersInput) {
+  const query = useInfiniteQuery<ListEventAttendeesRes.AsObject, GrpcError>({
+    queryKey: eventAttendeesKey({ eventId, type }),
+    queryFn: () => service.events.listEventAttendees({ eventId, pageSize }),
+    getNextPageParam: (lastPage) => lastPage.nextPageToken || undefined,
+  });
+  const attendeesIds =
+    query.data?.pages.flatMap((data) => data.attendeeUserIdsList) ?? [];
+  const { data: attendees, isLoading: isAttendeesLoading } =
+    useUsers(attendeesIds);
+
+  return {
+    ...query,
+    attendeesIds,
+    attendees,
+    isLoading: query.isLoading || isAttendeesLoading,
+  };
```

Comment:
I'm not sure about including `useUsers` here, because it is tying user loading to attendee loading. As a result, this query can revert back to `isLoading` if the new list has not-loaded users, which doesn't seem correct to me - once loaded, a query should only ever be fetching if it's stale.

On the other hand, I only caught this because of the backend bug so maybe it doesn't matter. But I don't see why not just use `useUsers` where it is needed instead of bundling it.

Reply:
Good point. Although the full user objects are kinda related to the attendees query though, as in, we can't display anything in the UI without both pieces. So for organisation purposes, I prefer to have it as part of this custom hook. 

Maybe I can return the loading states (`isAttendeesLoading` and `isUsersLoading` or something similar) separately to reduce the problem you mentioned there, as in the case of more new users being added, I don't think we can avoid not having both hooks going back to the loading state?

Reply:
In the case of new users added, it should remain "fetching" without changing what is displayed until the new users are loaded

Reply:
I'm not sure there's an easy way to tell whether something is refetching at the moment with `useUsers`, since I believe `isLoading` and `isFetching` will both be true regardless of if it's on first load or on refetch. Might have to refactor the hook slightly to compute this

Reply:
Oh I lied, apparently `isLoading` remains false if data exists already, should be easier than expected then hopefully 🤞